### ISSUE TITLE
E275: whitespace after annotated `match` variable (2.8.0 release blocker)

### ISSUE THREAD

**Issue by asottile** [PyCQA/pycodestyle#1024]
this code was previously ok, but is now triggering E275 on the `match` line:

```python
class Found(NamedTuple):
    y: int 
    match: Match[str]
```

**Comment by asottile** [comment]
this is also getting triggered for:

```python
def f():
    matcher = {"true": True, "false": False}.__getitem__
    #    ^
```

**Comment by cdce8p** [comment]
Thanks for bringing this up! Didn't had time to look at it in detail just yet, but I think the regex pattern was missing some end of line lookahead for it to work:
```diff
- ^\s*\b(?:match|case)(\s*)(?=.*\:)
+ ^\s*\b(?:match|case)(\s*)(?=.*\:(?=\s*$|\s*#))
```

**Comment by asottile** [comment]
that doesn't work either which is why I chose to revert rather than fix

```python
match 0:
    case _: print('hi')
```

**Comment by cdce8p** [comment]
The new pattern would at least not create false-positives, although it couldn't detect an error in your example either.
Not sure there is an easy regex for all possible cases. `match` and `case` only being soft keywords complicates things a bit. Combine that with the ability to nest arbitrary patterns.

**Comment by cdce8p** [comment]
Opened #1033 with a second attempt at adding the whitespace checks. I've taken into account the previous comments and believe I've even found a solution to the inline `case`.
